<?xml version="1.0" encoding="UTF-8"?>
<launch>

    <arg name="radius" default="2.0"/>
    <arg name="yaw_offset" default="0.0"/>
    <param name="use_sim_time" value="true"/>
    
    
    <!-- Robot 1 -->
    <arg name="ns1" default="robot1"/>
    <arg name="x1" value="$(eval cos(0) * arg('radius'))" />
    <arg name="y1" value="$(eval sin(0) * arg('radius'))" />
    <arg name="z1" default="0.0" />
    <arg name="yaw1" value="$(eval pi/2 + arg('yaw_offset'))" />
    
    <arg name="include_aruco_marker1" default="true"/>
            <arg name="robot_model1" default="$(find p3dx_description)/urdf/pioneer3dx.xacro"/>            
    <group ns="$(arg ns1)">
        <param name="tf_prefix" value="$(arg ns1)_tf" />
        <include file="$(find p3dx_gazebo)/launch/spawn.launch">
            <arg name="name" value="$(arg ns1)" />
            <arg name="x" value="$(arg x1)" />
            <arg name="y" value="$(arg y1)" />
            <arg name="z" value="$(arg z1)" />
            <arg name="yaw" value="$(arg yaw1)" />
            <arg name="namespace_arg" value="$(arg ns1)" />
            
            <arg name="robot_model" value="$(arg robot_model1)"/>
            

        </include>
      
    </group>
    
</launch>
